# 🛒 Market Basket Analysis - Proyecto Final de Minería de Datos

Este proyecto tiene como objetivo aplicar técnicas de minería de datos para identificar patrones de compra en un conjunto de transacciones de supermercado. Para ello, se utiliza el **Groceries Dataset**, el cual contiene información sobre clientes, fechas de compra y productos adquiridos.

El análisis se desarrolla mediante reglas de asociación, utilizando principalmente el algoritmo **Apriori** y comparándolo con **FP-Growth**. A partir de estos algoritmos se calcularán métricas como **support**, **confidence** y **lift**, que permiten descubrir qué productos suelen comprarse juntos.

Este notebook corresponde a la fase inicial de **comprensión de los datos** dentro de la metodología **CRISP-DM**, donde se explora la estructura del dataset antes de realizar la limpieza, transformación y modelado.

## 1. Carga y revisión inicial del dataset

En esta sección se carga el archivo `Groceries_dataset.csv` y se revisan sus primeras filas, dimensiones, tipos de datos y cantidad de productos y clientes únicos.

In [1]:
import pandas as pd

df = pd.read_csv('../data/raw/Groceries_dataset.csv')
print(f"Filas: {df.shape[0]}, Columnas: {df.shape[1]}")
print(df.head())
print(df.info())
print(f"\nProductos únicos: {df['itemDescription'].nunique()}")
print(f"Clientes únicos: {df['Member_number'].nunique()}")

Filas: 38765, Columnas: 3
   Member_number        Date   itemDescription
0           1808  21-07-2015    tropical fruit
1           2552  05-01-2015        whole milk
2           2300  19-09-2015         pip fruit
3           1187  12-12-2015  other vegetables
4           3037  01-02-2015        whole milk
<class 'pandas.DataFrame'>
RangeIndex: 38765 entries, 0 to 38764
Data columns (total 3 columns):
 #   Column           Non-Null Count  Dtype
---  ------           --------------  -----
 0   Member_number    38765 non-null  int64
 1   Date             38765 non-null  str  
 2   itemDescription  38765 non-null  str  
dtypes: int64(1), str(2)
memory usage: 1.6 MB
None

Productos únicos: 167
Clientes únicos: 3898


## Interpretación inicial del dataset

Al cargar el dataset se observa que contiene **38.765 registros** y **3 columnas**: `Member_number`, `Date` e `itemDescription`.

La columna `Member_number` identifica al cliente, `Date` representa la fecha de compra y `itemDescription` indica el producto adquirido. No se evidencian valores nulos en esta revisión inicial.

Además, el dataset cuenta con **167 productos únicos** y **3.898 clientes únicos**, lo que permite construir transacciones agrupando los productos comprados por cada cliente en una misma fecha. Esta estructura es adecuada para aplicar técnicas de reglas de asociación como **Apriori** y **FP-Growth**.